### ISSUE TITLE
1.7.10 Improvements

### ISSUE THREAD

**Issue by kmecpp** [MachineMuse/MachineMusePowersuits#898]
Multiple players have reported losing all energy in their power armor after teleporting

Also I don't think creative players should not lose energy or gain heat

I can submit a PR soon-ish if you're no longer supporting 1.7.10

**Comment by lehjr** [comment]
Heat and drain are left on for development reasons as these things are often debugged in creative mode. Since this version is no longer under active development, they can probably be disabled for creative mode. Teleportation drain is an old bug related to the sprint assist module and was fixed in 1.10.2. Backporting the fix should be pretty simple. I'll look at it tomorrow.

**Comment by kmecpp** [comment]
Thanks. Might be best to have a config option for creative heat drain in the updated versions then

**Comment by lehjr** [comment]
Not really a priority at the moment since the only supported version is still alpha and under development.

**Comment by lehjr** [comment]
Fixing the power drain issue was relatively easy. The creative "bypass" is another matter due to the implementation of multiple power systems and multiple points are where power drain is called, all lacking a player parameter.  Without a player parameter, there is no way to check if the player is in creative mode.

**Comment by lehjr** [comment]
ElectricItemUtils.drainPlayerEnergy is about the only place this can be implemented. Keep in mind this isn't likely to be 100% effective. 
The story for the heat system is similar. I've implemented a creative check on both of those where they can be applied, but again, it's not likely to be 100% reliable.

**Comment by kmecpp** [comment]
Understood. I think anything to make it generally less annoying would be quite helpful.

Also thanks for fixing the teleporting issue! Will you post the changes soon?

**Comment by lehjr** [comment]
probably later today. I was looking into fixing a SmartMoving bug, but unfortunately the mod author's MinecraftForums.net account and forums posts fell victim to the purge where roughly 1 million users and 10 million posts got deleted due to the take over by Fandom. That alone wouldn't be an issue, but the only documentation and source for the mods were posted there.

**Comment by eyeonus** [comment]
Have you tried the wayback machine?

http://web.archive.org/web/20181120050755/https://www.minecraftforum.net/forums/mapping-and-modding-java-edition/minecraft-mods/1274224-smart-moving

**Comment by eyeonus** [comment]
Also, here's the DropBox for the last version, which has the source code inside the .zip in the "development" sub-folder.

https://www.dropbox.com/sh/w2qf6kej868412r/AAA5SuRdaftT2p0CPJ7R3xmLa?preview=MC+1.8.9+-+Smart+Moving+16.3.zip

And the Dropbox folder that has, apparently, every version of SmartMoving ever.

https://www.dropbox.com/sh/w2qf6kej868412r/AAA5SuRdaftT2p0CPJ7R3xmLa

**Comment by lehjr** [comment]
Thanks. Unfortunately, the whole thing is
SmartMoving
SmartRenderer
RenderPlayerAPI
PlayerAPI

For 1.12.2, I did fix the model, but untested due to coremod signing requirements and no compiled version being available.

**Comment by eyeonus** [comment]
The wayback machine is very useful.

(SmartRenderer's source is in with the SmartMoving files, in the same "development" sub-folder.)

RenderPlayerAPI:
http://web.archive.org/web/20180512055645/https://www.minecraftforum.net/forums/mapping-and-modding-java-edition/minecraft-mods/1283669-render-player-api

https://www.dropbox.com/sh/2nfhgvo4nh037xp/AADeghcvD5uON-c-uFuGUL6Ja

PlayerAPI:
http://web.archive.org/web/20180927134817/https://www.minecraftforum.net/forums/mapping-and-modding-java-edition/minecraft-mods/1277996-player-api

https://www.dropbox.com/sh/wfd5j8p3hp6rhmd/AACRgjxNT176S1cma_MijD_Fa

**Comment by lehjr** [comment]
Thanks. I'm not entirely sure yet, but I'm relatively certain the issue is in one of the 4 and not in MPS. So I could probably release what I have and if I find a fix, release that separately, though there isn't any official channels for that

**Comment by eyeonus** [comment]
Divisor does still have a CurseForge account, so maybe a pm sent via that?

https://minecraft.curseforge.com/members/Divisor

**Comment by lehjr** [comment]
Yeah, forgot about that. That might work. There seems to be some recent activity on that account too.

**Comment by eyeonus** [comment]
It's too bad Divisor never GitHub'd his stuff....

**Comment by lehjr** [comment]
I know, it would have probably gotten more fixes for issues.

**Comment by eyeonus** [comment]
I've half a mind to take his stuff and GitHub it myself, but I have no desire to work on it, so that'd be a bit pointless. I mean, I could give contributor access to other people, but then again, other people could just make their own fork....

**Comment by lehjr** [comment]
It's one of those things where it's not really worth the time unless you're both willing and able to support it. I'm going to try to fix this one issue, but I want no part in maintaining that project.

**Comment by kmecpp** [comment]
> Fixing the power drain issue was relatively easy.

Was the fix for this ever commited?

**Comment by lehjr** [comment]
> > Fixing the power drain issue was relatively easy.
> 
> Was the fix for this ever commited?
Should have been, but who knows, Either way, 1.17.10 is years old and I'm not working on it. I already have my hands full with 1.14.4.

**Comment by kmecpp** [comment]
Could you just check your 1.7.10 branch and see if you have the uncommited patch sitting there?

That comment where you said you fixed it was from Jan 27 and the latest commit to the branch is Jan 17

**Comment by lehjr** [comment]
Whether it is or isn't won't change whether or not I work on it or not. If it didn't get committed then the code is gone.

**Comment by lehjr** [comment]
Looks like you're in luck after all since it was part of the backup I made before wiping my primary OS.

**Comment by kmecpp** [comment]
Awesome, thank you!

**Comment by lehjr** [comment]
And that's the end of 1.7.10 support from me.

**Comment by MalTeeez** [comment]
Im sorry if this sounds dumb, but was it ever released?

**Comment by MalTeeez** [comment]
I noticed the commit in 1.7.10 but saw a direct release - Im not good in building it myself

**Comment by lehjr** [comment]
was probably released months ago, but keep in mind that 1.17.10 is not the only version of the mod, so I can't keep track of them all.

**Comment by MalTeeez** [comment]
Where could it have been released, I looked on both Curse Pages and the release Section - Is there something else?

**Comment by lehjr** [comment]
No that's it. Unfortunately, setting up a 1.7.10 build environment is not trivial, especially for this mod. And I simply do not have the time at the moment. This is a team of 1, and I'm already working on 1.14.4 and 1.12.2. I cannot split my time any further than that.

**Comment by lehjr** [comment]
Having said all that, Modular Powersuits and Numina are both more or less a dead end but will live on as Modular Power Armor and MPA-Lib. I will be providing a port of Modular Powersuits to use MPA-Lib in 1.12.2, but 1.7.10 is not currently planned. 

https://github.com/MachineMuse/MachineMusePowersuits/issues/936#issuecomment-544188795

**Comment by MalTeeez** [comment]
I acknowledged that, was just wondering if it ever got released since I couldn't find it
But is there a way I can build it myself since the 1.7.10 Branch is still there

**Comment by lehjr** [comment]
if it was that trivial I would have done it already. That said, you're welcome to try

**Comment by MalTeeez** [comment]
@kmecpp @eyeonus Do you a working build?

**Comment by kmecpp** [comment]
I do but it has many other changes as well. I don't remember it being hard to setup the work space for this mod. Try just running

gradlew setupDecompWorkspace --refresh-dependencies
gradlew idea
gradlew genIntellijRuns
gradlew build

**Comment by MalTeeez** [comment]
Ill try that, thanks. BTW. what other changes did you come up with?

**Comment by MalTeeez** [comment]
Nope didn't work
